distributed/shuffle/_arrow.py

```diff
@@ -54,7 +54,9 @@ def convert_partition(data: bytes) -> pa.Table:
     while file.tell() < end:
         sr = pa.RecordBatchStreamReader(file)
         shards.append(sr.read_all())
-    return pa.concat_tables(shards)
+    table = pa.concat_tables(shards)
+    df = table.to_pandas(self_destruct=True)
```

Comment:
I took it as a recommendation from https://arrow.apache.org/docs/python/pandas.html#reducing-memory-use-in-table-to-pandas

From what I understand, it would wreck havoc if we used `table` after the call to `to_pandas`. Given that we return after the next line, that shouldn't be a problem.

Reply:
Thanks for pointing to the extra docs. I guess I was nervous about `pyarrow`-backed dtypes in `pandas` specifically. But I would hope in that case memory wouldn't be deallocated if `pandas` was using it after the `pyarrow` -> `pandas` handoff. If we're running this against a few use cases that do things after a shuffle (which is sounds like we are) then we should have sufficient coverage